src/properties/components/Properties.tsx

```diff
@@ -50,28 +58,22 @@ export const Properties = () => {
     () => tryFindByUUID(uiSchema, selection?.uuid),
     [selection, uiSchema]
   );
-  const canSetUISchemaOptions = (
-    uiElement: EditorUISchemaElement | undefined,
-    updatedProperties: any
-  ): boolean =>
-    !!uiElement &&
-    !isEqual(updatedProperties, uiElement.options) &&
-    !(uiElement.options === undefined && isEmpty(updatedProperties));
 
   const updateProperties = useCallback(
     ({ data: updatedProperties }) => {
-      if (canSetUISchemaOptions(uiElement, updatedProperties)) {
+      if (canUpdateUiSchemaElement(uiElement, updatedProperties)) {
         dispatch(Actions.setUiSchemaOptions(uiElement, updatedProperties));
       }
     },
     [dispatch, uiElement]
   );
+
   if (!selection) return <NoSelection />;
 
   const properties = getProperties(uiElement, schema);
   return properties ? (
     <JsonForms
-      data={uiElement.options}
+      data={uiElement}
```

Comment:
Here we should give the mentioned extracted element. The extracted element can be created using lodash's omit, e.g. `omit(uiSchemaElement, ['uuid', 'parent', 'elements' 'options.detail']`.

Reply:
can't we do something like `const {uuid, parent, elements, options.detaill, ...base} = uiSchemaElemnt;` ?

Reply:
I guess that approach would work but it would look slightly more complicated. I quickly tried it and this is what I came up with:
```typescript
const canUpdateUiSchemaElement = (
  uiElement: any,
  updatedProperties: any
): boolean => {
  return !isEqual(uiElement, updatedProperties);
};

export const Properties = () => {
  const [selection] = useSelection();
  const uiSchema = useUiSchema();
  const schema = useSchema();
  const dispatch = useDispatch();

  const uiElement: EditorUISchemaElement = useMemo(
    () => tryFindByUUID(uiSchema, selection?.uuid),
    [selection, uiSchema]
  );

  const extractedElement = useMemo(() => {
    if (!uiElement) {
      return undefined;
    }
    const {
      uuid,
      linkedSchemaElement,
      parent,
      elements,
      options,
      ...base
    } = uiElement as EditorLayout;
    return {
      ...base,
      options: options && { ...options, detail: undefined },
    };
  }, [uiElement]);

  const updateProperties = useCallback(
    ({ data: updatedProperties }) => {
      if (canUpdateUiSchemaElement(extractedElement, updatedProperties)) {
        dispatch(Actions.setUiSchemaOptions(uiElement, updatedProperties));
      }
    },
    [dispatch, extractedElement, uiElement]
  );
// [...]
```